### ISSUE TITLE
wrappers for python and PyPy

### ISSUE THREAD

**Issue by mattip** [swig/swig#1645]
We have gotten two reports of failures with swig python wrappers and PyPy. Both involve a call to a function with starargs producing a "argument after * must be an iterable, not None" error. One is in [TensorFlow](https://github.com/tensorflow/tensorflow/issues/252), the other in [CVXPY](https://stackoverflow.com/questions/58328323/using-cvxpy-under-pypy3). I am a PyPy dev, but not familiar with the swig testing framework. How would I go about running a test suite on PyPy that might expose the problem?

**Comment by vadz** [comment]
I haven't actually tried it, but you should be able to just do `make check-python-test-suite PYBIN=pypy`.

**Comment by mattip** [comment]
Starting off simple, this failed to run the test suite with python3
```
source cpython3_virt/bin/activate
./configure --with-python3
make clean; make
SWIGLANG=python PY3=3 make check-python-test-suite
```

It complains that it cannot find any of the `*_runme3.py` files to run the tests:
```
checking python test-suite
checking python testcase argcargvtest (with run test)
env: ‘argcargvtest_runme3.py’: No such file or directory
Makefile:119: recipe for target 'argcargvtest.cpptest' failed
make[1]: *** [argcargvtest.cpptest] Error 127
```

**Comment by mattip** [comment]
my mistake, I was missing `pip install 2to3`. Now the test suite runs with python3 after activating a virtualenv: `PY3=3 make check-python-test-suite`

**Comment by mattip** [comment]
I reset the virtualenv to pypy3, and could run the test after tweaking the `Examples/Makefile` (perhaps if I reran ./autogen.sh I would not have needed this)
-  `PYTHON_INCLUDE= $(DEFS) -I/<path-to-pypy>/include`
- `PYTHON_SO     = .pypy3-72-x86_64-linux-gnu.so`

Unfortunately none of the [test failures](https://gist.github.com/mattip/3eb6c10bb722f2f3583f135342e1e548) have the error message I was looking for, they are mostly about refcount failures.

**Comment by mattip** [comment]
The problem is that the package is using files generated by SWIG 3.0.8, the bug where `SWIG_Python_NewShadowInstance` would call `tp_new(... Py_None, Py_None)` was fixed in 0165180735 which was part of the 4.0.0 release
